pkg/ee/resource-quota/handler.go

```diff
@@ -375,5 +375,17 @@ func convertToCRDQuota(quota apiv2.Quota) (kubermaticv1.ResourceDetails, error)
 		return kubermaticv1.ResourceDetails{}, fmt.Errorf("error parsing quota Storage %w", err)
 	}
 
-	return *kubermaticv1.NewResourceDetails(cpu, mem, storage), nil
+	resourceDetails := kubermaticv1.NewResourceDetails(cpu, mem, storage)
+
+	if resourceDetails.CPU.IsZero() {
+		resourceDetails.CPU = nil
+	}
+	if resourceDetails.Memory.IsZero() {
+		resourceDetails.Memory = nil
+	}
+	if resourceDetails.Storage.IsZero() {
+		resourceDetails.Storage = nil
+	}
+
+	return *resourceDetails, nil
```

Comment:
This seems a bit hackish to me. Shouldn't we just treat 0 as nil? For example, If I create the resource quota directly via CR and give `CPU` the value `0`. Then I'll end up with the same issue, right?

Reply:
> If I create the resource quota directly via CR and give CPU the value 0. Then I'll end up with the same issue, right?

Yes. It actually may be a good idea as there is no value in keeping 0 different than nil unless we want to allow to blocking resource creation inside the project. But at this stage I don't want to risk changing CR and the way controllers interpret zero value  (cc @lsviben )

I'm currently testing changing values to pointers in the API model and replacing PATCH with PUT as an alternative solution.

Reply:
I agree with you, for now, let's just settle with this.

Reply:
I've replaced values with pointers and to make the API consistent with the behaviour under the hood introduced PUT in place of PATCH. I'll get rid of PATCH soon, but first I want to update the Dashboard, hence it got deprecated for now. Please take a look @ahmedwaleedmalik if possible.

Reply:
oh, I see you already taken a glance :D

Reply:
Yes :D